http/src/main/java/io/crate/plugin/PipelineRegistry.java

```diff
@@ -73,6 +92,7 @@ public void registerItems(ChannelPipeline pipeline) {
         for (PipelineRegistry.ChannelPipelineItem item : addBeforeList) {
             pipeline.addBefore(item.base, item.name, item.handlerFactory.get());
         }
+        sslHandler.addToPipeline(pipeline);
```

Comment:
Fair point. I think we can remove the default implementation and pass the PipelineRegistry via the constructor arg of the ssl implementation.